src/ensembl/src/content/app/App.tsx

```diff
@@ -14,99 +14,80 @@
  * limitations under the License.
  */
 
-import React, { useEffect, FunctionComponent, lazy, Suspense } from 'react';
-import { Route, RouteComponentProps, Switch } from 'react-router-dom';
+import React, { useEffect, lazy, Suspense } from 'react';
+import { Route, Switch, useLocation } from 'react-router-dom';
 import { connect } from 'react-redux';
 
 import { changeCurrentApp } from 'src/header/headerActions';
-import { getCurrentApp } from 'src/header/headerSelectors';
 
 import ErrorBoundary from 'src/shared/components/error-boundary/ErrorBoundary';
 import { NewTechError } from 'src/shared/components/error-screen';
 
-import { RootState } from 'src/store';
-
+const HomePage = lazy(() => import('../home/Home'));
 const GlobalSearch = lazy(() => import('./global-search/GlobalSearch'));
 const SpeciesSelector = lazy(() =>
   import('./species-selector/SpeciesSelector')
 );
+const SpeciesPage = lazy(() => import('./species/SpeciesPage'));
 const CustomDownload = lazy(() => import('./custom-download/CustomDownload'));
 const Browser = lazy(() => import('./browser/Browser'));
 const EntityViewer = lazy(() => import('./entity-viewer/EntityViewer'));
 
-type StateProps = {
-  currentApp: string;
-};
-
-type DispatchProps = {
+type AppProps = {
   changeCurrentApp: (name: string) => void;
 };
 
-type OwnProps = {};
-
-type AppProps = RouteComponentProps & StateProps & DispatchProps & OwnProps;
-
 type AppShellProps = {
   children: React.ReactNode;
 };
 
 export const AppShell = (props: AppShellProps) => {
-  return <>{props.children}</>;
+  return <div>{props.children}</div>;
 };
 
 const AppInner = (props: AppProps) => {
-  const { url } = props.match;
+  const location = useLocation();
 
   useEffect(() => {
-    // remove /app/ from url to get app name
-    let appName = props.location.pathname.replace('/app/', '');
-
-    // check if app name still has forward slash (/) to be sure the app name is extracted
-    // if it isn't then remove rest of the URL and extract the app name
-    if (appName.indexOf('/') > -1) {
-      const matches = appName.match(/^[^\/]*/);
-      appName = matches ? matches[0] : '';
-    }
+    const appName: string = location.pathname.split('/').filter(Boolean)[0];
```

Comment:
Do we need `filter(Boolean)` here?

Reply:
`"/foo/bar".split('/')` returns `["", "foo", "bar"]`. I have not checked whether `location.pathname` will start with a slash, but I expect it will.

On a different note, I doubt we need to keep the app name in redux at all — I don't remember us doing anything useful with it. If needed, the app name can trivially be derived from the url. It's probably worth exploring whether we want to keep this function at all — but not in this PR.